### PR TITLE
FileSystemServiceDiscovery

### PR DESCRIPTION
**Pull request by yiduwangkai**
**What this PR does**:
FileSystemServiceDiscovery, the implementation of ServiceDiscovery based on file system.

**Which issue(s) this PR fixes**:
#426 FileSystemServiceDiscovery 
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: registry/filesystem/file.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"os"
+)
+
+type File struct {
+	Path         string   //路径
+	fileWriter   *os.File //写入文件
+}
+
+// 创建文件锁，配合 defer f.Release() 来使用
+func (file *File)Create() (f *Filelock, e error) {
```
Comment (AlexStocks) [inline registry/filesystem/file.go:31]: hey, guy, there are two improvements to do. Numnber 1, pls do not use Chinese in comment. Number 2, pls move the fundamental funcs to github.com/gost. 

as this func, pls submit it to the github.com/gost/os. and if this dir do not exist, pls create it.

File: registry/filesystem/file.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	if file == nil || file.Path == "" {
+		return nil, perrors.Errorf("cannot create flock on empty path")
+	}
+	lock, e := os.Create(file.Path)
```
Comment (hxmhlt) [inline registry/filesystem/file.go:35]: len(file.Path)==0 来判断字符串是否为空更好

File: registry/filesystem/file_lock.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filesystem
+
+import (
+	"github.com/apache/dubbo-go/common/logger"
```
Comment (hxmhlt) [inline registry/filesystem/file_lock.go:21]: import需要分段，看下其他文件的依赖导入格式吧~

File: registry/filesystem/file.go (hunk 1 above)
Comment (hxmhlt) [reply registry/filesystem/file.go:31]: 每一个func前都需要添加注释，格式为
// Create(the first word have to be same as your func name)...

Comment (flycash) [reply registry/filesystem/file.go:31]: it would be better to use English. ：）

Comment (yiduwangkai) [reply registry/filesystem/file.go:31]: OK

File: registry/filesystem/service_discovery.go [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+	configCenterDirParamName = paramNamePrefix + "dir"
+	configCenterEncodingParamName = paramNamePrefix + "encoding"
+	defaultConfigCenterEncoding = "UTF-8"
+)
+
+func init() {
+	extension.SetServiceDiscovery(name, newFileSystemServiceDiscovery)
+}
+
+func newFileSystemServiceDiscovery(url *common.URL) (registry.ServiceDiscovery, error){
```
Comment (flycash) [inline registry/filesystem/service_discovery.go:50]: Should it be a singleton?

File: registry/filesystem/service_discovery.go [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+func exists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil { return true, nil }
+	if os.IsNotExist(err) { return false, nil }
+	return true, err
+}
+
+// FilleSystemServiceDiscovery is an implementation based on memory.
```
Comment (flycash) [inline registry/filesystem/service_discovery.go:99]: based on file system

File: registry/filesystem/service_discovery.go [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+	encoding       string
+	rootDirectory  *File
+}
+
+func (i *FilleSystemServiceDiscovery) String() string {
+	return name
+}
+
+// Destroy doesn't destroy the instance, it just clear the instances
+func (i *FilleSystemServiceDiscovery) Destroy() error {
```
Comment (flycash) [inline registry/filesystem/service_discovery.go:113]: I think it should remove the files.

File: registry/filesystem/service_discovery.go [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+	// reset to empty
+	i.fileLocksCache = nil
+	i.listeners = nil
+	return nil
+}
+
+// Register will store the instance using its id as key
+func (i *FilleSystemServiceDiscovery) Register(instance registry.ServiceInstance) error {
+	return nil
```
Comment (flycash) [inline registry/filesystem/service_discovery.go:122]: Do nothing?

**Comment by AlexStocks** [comment]
will some guy accept this task to complete it?

**Pull request closed without merging**